sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,111 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import DVR
+
+public extension Filter {
+    
+    static let subscriptionIDRegex = "(/(subscriptions))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+    static let graphSubscriptionIDRegex = "(https://(graph.windows.net))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+    
+    static var subcriptionIDReplacement = "99999999-9999-9999-9999-999999999999"
+    
+    static var standardHeaderScrubbing: [String : FilterBehavior] = [
+        "location": .closure({ key, val in
+            scrubSubscriptionId(from: val ?? "")
+        }),
+        "operation-location": .replace(""),
+        "azure-asyncoperation": .replace(""),
+        "www-authenticate": .replace(""),
+        "access_token": .replace("")
+        
+    ]
+    
+    static func scrubSubscriptionId(from string: String?) -> String? {
```

Comment:
Even though this is an internal method, doc comments would still be helpful for intellisense purposes.

Reply:
I think this would define a much clearer contract if the signature were:
```swift
/// Returns the original string with subscription ID info (if any) replaced.
func scrubSubscriptionId(from string: String) -> String {...}
```

Many of my comments stem from seeing this invocation in the wild with no clear understanding of why it could return `nil`.

Reply:
It can return nil because URLRequest's url and a few others are optional strings. It would return nil in the if it's passed something nil.

Reply:
Yep, I get that. If the only reason it will return nil is if it is passed nil, it should simply not be allowed to pass nil. While it can be a convenience (which we used in Swift APIView, for instance) in this case it is for scrubbing sensitive information. Allowing the method to return nil invites real or perceived vulnerabilities.

Reply:
Would you prefer I overload this function with strictly a String parameter and return type?

Reply:
Not an overload--just remove the optionality in the parameter and return type.

Reply:
Im gonna make a comment about this on the updated version